Source/VisualStudioTools/Private/VSTestAdapterCommandlet.cpp

```diff
@@ -201,10 +225,36 @@ int32 UVSTestAdapterCommandlet::Main(const FString& Params)
 		return 0;
 	}
 
-	FAutomationTestFramework::GetInstance().SetRequestedTestFilter(EAutomationTestFlags::PriorityMask |
-		EAutomationTestFlags::ProductFilter | EAutomationTestFlags::SmokeFilter |
-		EAutomationTestFlags::PerfFilter |
-		EAutomationTestFlags::StressFilter | EAutomationTestFlags::NegativeFilter);
+    // Default to all the test filters on except for engine tests.
+    uint32 filter = EAutomationTestFlags::PriorityMask |
+        EAutomationTestFlags::ProductFilter | EAutomationTestFlags::SmokeFilter |
+        EAutomationTestFlags::PerfFilter | EAutomationTestFlags::StressFilter | EAutomationTestFlags::NegativeFilter;
+    if (ParamVals.Contains(SmokeFilterParam) && ParamVals[SmokeFilterParam] == "False")
```

Comment:
As we've talked over chat, we might want to avoid the multiple params and combine this into a single flag.
To separate them, you can use a `+` sign like below. Then the code just do a split or even just string contains.

Note that this saves you to have to do case insensitive matching on that "True"/"False" strings, which should be the case, because otherwise someone decides to lowercase the command in the VS code and things break.
```
 -filters="engine+product+smoke+stress+perf"
```